src/Compilers/CSharp/Portable/Syntax/SyntaxNodeRemover.cs

```diff
@@ -172,8 +190,16 @@ private bool ShouldVisit(SyntaxNode node)
                 {
                     if (this.IsForRemoval(node))
                     {
-                        this.AddTrivia(node);
-                        result = null;
+                        if (IsEmbeddedStatement(node))
+                        {
+                            // Embedded statements cannot be null.
+                            result = SyntaxFactory.Block().WithTriviaFrom(node);
+                        }
+                        else
+                        {
+                            this.AddTrivia(node);
+                            result = null;
+                        }
```

Comment:
📝 This change should no longer be necessary.

Reply:
After merging #46666, the tests from this PR fail with the same ArgumentNullException, so I believe this change is still necessary.

Reply:
i'm not a fan of SynaxNodeRemover having this change.  It feels out of place for it to know about this.  You could certainly update the remove-unreachable-code fixer to make thsi sort of change though.

Reply:
Not a fan either. But the alternative is to include a similar piece of code in every fixer that might deal with embedded statements. There are two already: "remove unused variable" and "remove unreachable code"; and there might be more.
The rule "embedded statements cannot be null" feels like a part of domain description present at the compiler level, so it's a little strange that manipulations with the code using SyntaxNodeRemover can violate this rule.

Reply:
> Not a fan either. But the alternative is to include a similar piece of code in every fixer that might deal with embedded statements.

Yes.  I think each feature should have to decide how they want to handle this case.  I don't see this as a general-purpose solution.

> so it's a little strange that manipulations with the code using SyntaxNodeRemover can violate this rule.

SyntaxNodeRemover can break every invaraint in the compiler.  It allows any node to be removed, and that can certain crash.  I don't see why we woudl special case thsi case while you can still run into this with every other required node in teh syntax model.

Reply:
Agreed; I moved the change to the code fixer.